runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -194,6 +195,9 @@ public void write(@Nonnull IToken token, @Nonnull Object data, @Nonnull CacheOpt
             ld = new LogData(DataType.DATA, data);
         }
 
+        // This write method should only accept LogData without the token and Object types.
+        ld.setPayloadCodecType(Codec.Type.valueOf(runtime.getParameters().getCodecType()));
```

Comment:
Compression is leaking into `AddressSpace`. Why not set the codec type during `ILogData` creation? Can't we make `CorfuRuntimeParameters` be a mandatory parameter for `ILogData`?

Reply:
Good point, perhaps its cleaner just to make mandatory the codec type, as all other CorfuRuntimeParameters are irrelevant to LogData.

Reply:
+1